BAT/BCEngineMCMC.h

```diff
@@ -272,34 +289,43 @@ class BCEngineMCMC
     const std::vector<std::vector<double> >& GetScaleFactors() const
     { return fMCMCProposalFunctionScaleFactor; }
 
+    /**
+     * @return states of all chains */
+    const std::vector<ChainState>& GetChainStates() const
+    { return fMCMCStates; }
+
+    /**
+     * @param c index of the Markov chain
+     * @return state of chains */
+    const ChainState& GetChainStates(unsigned c) const
```

Comment:
If we have `GetChainStates()`, we don't need `GetChainStates(unsigned c)` in my opinion

Reply:
Actually the opposite is better. Let's remove `GetChainStates()` and keep `GetChainState(unsigned c)`. Note the missing `s` on `...State`. This gives us the opportunity to change the underlying container if we wanted to.

Reply:
I was just making it match to `Getx()` and `GetProb()`. I don't think we should have those either now. But for back-compatibility I left them in.